internal/webhook/v1beta1/registrycacheconfig_webhook.go

```diff
@@ -66,34 +74,48 @@ func (v *RegistryCacheConfigCustomValidator) ValidateCreate(_ context.Context, o
 	}
 	registrycacheconfiglog.Info("Validation for RegistryCacheConfig upon creation", "name", registrycacheconfig.GetName())
 
-	// TODO(user): fill in your validation logic upon object creation.
+	var registrycacheconfigs corekymaprojectiov1beta1.RegistryCacheConfigList
+	err := v.client.List(context.Background(), &registrycacheconfigs, client.InNamespace(registrycacheconfig.Namespace))
+	if err != nil {
+		return nil, fmt.Errorf("failed to list existing RegistryCacheConfig resources: %w", err)
+	}
 
-	return nil, errors.New("not implemented temporarily")
+	var secretList v1.SecretList
```

Comment:
Do we really need to read all all secrets from current namespace? I suppose there will be plenty of them. Maybe we can use a dedicated label to mark secrets used for keeping registry-cache-configs?

```golang
var secretList v1.SecretList
	if err := v.client.List(context.Background(), &secretList,
		client.InNamespace(newRegistryCacheConfig.Namespace),
		client.MatchingLabels(map[string]string{"app.kubernetes.io/part-of": "registry-cache-config"})
	); err != nil {
		return nil, fmt.Errorf("failed to list secrets with registry cache configurations: %w", err)
	}
```

Reply:
I will consider changing it to get.